lib/cfn-nag/util/enforce_reference_parameter.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/util/truthy.rb'
+
+# Returns false if the provided key_to_check is a no-echo parameter
+# without a default value; true otherwise.
+# Only applicable for a hash
+def insecure_parameter?(cfn_model, key_to_check)
+  return false unless key_to_check.is_a? Hash
+  return false unless key_to_check.key? 'Ref'
+
+  if cfn_model.parameters.key? key_to_check['Ref']
+    parameter = cfn_model.parameters[key_to_check['Ref']]
+    if truthy?(parameter.noEcho) && parameter.default.nil?
+      return false
+    end
+  end
+
+  true
```

Comment:
this is dead code?

Reply:
nvm - i see it isn't dead code... the default being !nil will fall through here.

Reply:
so small nit would either be to comment the fallthrough, or else make the conditions explicit like if X then Y else Z to make it clear what the branches are

Reply:
I've added more comments into this function to hopefully make things more clear from a quick glance.